sympy/functions/elementary/exponential.py

```diff
@@ -991,7 +993,10 @@ def coeff_exp(term, x):
             n = p.getn()
         _, d = coeff_exp(p, x)
         if not d.is_positive:
-            return log(a) + b*logx + Order(x**n, x)
+            if (log(a) + b*log(x)) == self:
+                return log(a) + b*logx
+            else:
+                return log(a) + b*logx + Order(x**(n + 1), x)
```

Comment:
Why `n + 1`?

Reply:
That's a mistake, that was actually masking another bug (value of `n` was changing in the method), which I have changed now.